ckanext/security/fanstatic/login_ajax.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+(function ($) {
+  var getLoginForm = function () {
+    return $('#mfa-login-form')
+  }
+
+  var showQRCode = function (loginState) {
+    $('#totp-secret').text(loginState.totpSecret)
+    $('#mfa-setup').show()
+
+    new QRious({
+      size: 250,
+      element: $('#qr-code-container')[0],
+      value: loginState.totpChallengerURI
+    })
+  }
+
+  var handleMfaValidation = function (validCode) {
+    if (validCode) {
+      getLoginForm().off('submit', validateLoginCredentials)
+      getLoginForm().submit()
```

Comment:
How does this result in different behaviour, such as being redirected somewhere, instead of spitting json back out at you?

Reply:
Detaching the submit handler means that the subsequent "submit" call does the full form POST to `/login_generic` which is a non-ajax controller action.

Reply:
Ohh the form itself has a diff action URL, gotcha :+1: